hw-model/src/lib.rs

```diff
@@ -584,10 +597,14 @@ pub trait HwModel {
 
     /// Toggle reset pins and wait for ready_for_fuses
     fn warm_reset(&mut self) {
-        // sw-emulator lacks support: https://github.com/chipsalliance/caliptra-sw/issues/540
         panic!("warm_reset unimplemented");
     }
 
+    /// Toggle reset/pwrgood pins and wait for ready_for_fuses
+    fn cold_reset(&mut self) {
+        panic!("cold_reset unimplemented");
```

Comment:
This panic doesn't seem to jive with the PR title "Hw model cold reset support"

Reply:
I think this is just re-factoring before the actual implementation. It should be named accordingly.

Reply:
Sorry for the confusion. I did just reply to Kor about this in an IM. This is the base implementation. The FPGA and verilator implementations override this. This is what you get on the SW emulator though (further explanation on that above).

I will be adding a note here to make this clearer.

Reply:
This has been added now